cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -174,10 +175,14 @@ public BodyConsumer createAppVersion(HttpRequest request, HttpResponder responde
                                          @PathParam("namespace-id") final String namespaceId,
                                          @PathParam("app-id") final String appId,
                                          @PathParam("version-id") final String versionId)
-    throws BadRequestException, NamespaceNotFoundException {
+    throws Exception {
 
     ApplicationId applicationId = validateApplicationVersionId(namespaceId, appId, versionId);
 
+    if (!applicationLifecycleService.updateAppAllowed(applicationId))  {
+      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "There already exists application version " +
+        "with no '-SNAPSHOT' suffix for the same application to be deployed");
```

Comment:
Let's not expose '`-SNAPSHOT` in the error message (since it is an implementation detail)